compiler/src/dotty/tools/dotc/core/unpickleScala2/Scala2Unpickler.scala

```diff
@@ -535,7 +535,15 @@ class Scala2Unpickler(bytes: Array[Byte], classRoot: ClassDenotation, moduleClas
                 true) &&
             // We discard the private val representing a case accessor. We only enter the case accessor def.
             // We do need to load these symbols to read properly unpickle the annotations on the symbol (see sbt-test/scala2-compat/i19421).
-            !flags.isAllOf(CaseAccessor | PrivateLocal, butNot = Method)
+            !flags.isAllOf(CaseAccessor | PrivateLocal, butNot = Method) &&
+            // We don't enter Value Classes' extension methods from pickles, but we do
+            // generate in the ExtensionMethods phase, at the same time as we do for Scala 3
```

Comment:
Suggested rephrasing:
```suggestion
            // Skip entering extension methods: they will be recreated by the ExtensionMethods phase.
```